ern-local-cli/src/commands/cauldron/batch.js

```diff
@@ -184,24 +185,48 @@ exports.handler = async function ({
           updateDependencyObj.version)
         cauldronCommitMessage.push(`- Update ${updateDependencyObj.basePath} native dependency version to v${updateDependencyObj.version}`)
       }
-      // Update MiniApps
-      for (const updateMiniAppObj of updateMiniAppsObjs) {
-        // Add the MiniApp (and all it's dependencies if needed) to Cauldron
-        await updateMiniAppObj.addToNativeAppInCauldron(napDescriptor, force)
-        cauldronCommitMessage.push(`- Update ${updateMiniAppObj.name} MiniApp version to v${updateMiniAppObj.version}`)
-      }
       // Add Dependencies
       for (const addDependencyObj of addDependenciesObjs) {
         // Add the dependency to Cauldron
         await cauldron.addContainerNativeDependency(napDescriptor, addDependencyObj)
         cauldronCommitMessage.push(`-Add ${addDependencyObj.toString()} native dependency`)
       }
+      // Update MiniApps
+      for (const updateMiniAppObj of updateMiniAppsObjs) {
+        cauldronCommitMessage.push(`- Update ${updateMiniAppObj.name} MiniApp version to v${updateMiniAppObj.version}`)
+      }
       // Add MiniApps
       for (const addMiniAppObj of addMiniAppsObjs) {
-        // Add the MiniApp (and all it's dependencies if needed) to Cauldron
-        await addMiniAppObj.addToNativeAppInCauldron(napDescriptor, force)
         cauldronCommitMessage.push(`-Add ${addMiniAppObj.packageDescriptor} MiniApp`)
       }
+
+      const miniAppsInCauldron = await cauldron.getContainerMiniApps(napDescriptor)
+      const nonUpdatedMiniAppsInCauldron = _.xorBy(updateMiniAppsDependencyPaths, miniAppsInCauldron, 'basePath')
+      let nonUpdatedMiniAppsInCauldronObjs = []
+      for (const nonUpdatedMiniAppInCauldron of nonUpdatedMiniAppsInCauldron) {
+        const m = await MiniApp.fromPackagePath(nonUpdatedMiniAppInCauldron)
+        nonUpdatedMiniAppsInCauldronObjs.push(m)
+      }
+
+      const nativeDependencies = await resolver.resolveNativeDependenciesVersionsOfMiniApps([...updateMiniAppsObjs, ...addMiniAppsObjs, ...nonUpdatedMiniAppsInCauldronObjs])
+      const cauldronDependencies = await cauldron.getNativeDependencies(napDescriptor)
+      const finalNativeDependencies = resolver.retainHighestVersions(nativeDependencies.resolved, cauldronDependencies)
+
+      const conflictingDependencies = nativeDependencies.pluginsWithMismatchingVersions
+      if (conflictingDependencies.length > 0) {
+        if (force) {
```

Comment:
minor: this block appears at multiple places, may be a contender for externalizing in utils? Again not absolute necessary, just if it makes sense.